packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -794,6 +794,20 @@ export function bindInstance(
 
 export const supportsHydration = true;
 
+// With Resources, some HostComponent types will never be server rendered and need to be
+// inserted without breaking hydration
+export function isHydratable(type: string, props: Props): boolean {
```

Comment:
This feels like an odd special case. I'm not we'll even be able to do Placements in the middle of a hydrating tree in a future architecture since nothing else does it. Seems like not worth it. I wonder if we should just insert an empty script or something although that probably gets treated as an inline script.

Maybe it's not worth it and we should just leave this to `importAsyncScript`. This doesn't work today.

Reply:
I don't think setting src works. Once the script loads changing the src does not cause the script to load. We could get more extreme and construct a new script src and inject it in place of the hydrated one but this is super messy.

I'm leaning towards importAsyncScripts is the way to dedupe but still get your onloads and then just emit these scripts in line with a marker attribute so we can exclude them as resources